### PR TITLE
gh-102873: logging.LogRecord docs: improve description of msg parameter

### LINKED ISSUE #102873: logging module LogRecord can accept non-str msg but it's not documented
# Documentation

In the documentation of the logging `LogRecord` it's mentioned that it accepts msg as str argument:
https://docs.python.org/3/library/logging.html#logging.LogRecord

But just below that it explains that in `getMessage`:

> Returns the message for this [LogRecord](https://docs.python.org/3/library/logging.html#logging.LogRecord) instance after merging any user-supplied arguments with the message. If the user-supplied message argument to the logging call is not a string, [str()](https://docs.python.org/3/library/stdtypes.html#str) is called on it to convert it to a string. This allows use of user-defined classes as messages, whose __str__ method can return the actual format string to be used.

So non-str messages are allowed, and also in LogRecord attributes it's mentioned that non-str messages work and are accepted:
https://docs.python.org/3/library/logging.html#logrecord-attributes

Initially started the discussion in https://github.com/python/typeshed/pull/9914

### Linked PRs
* gh-102875
* gh-103003
* gh-103004

### PR DESCRIPTION
**Pull request by aminalaee**
Closes https://github.com/python/cpython/issues/102873

* Issue: gh-102873

### CODE REVIEW

File: Doc/library/logging.rst
Comment (AlexWaygood) [inline Doc/library/logging.rst]: This is a good type hint for typeshed, where the primary consumer of the type hints is a type checker. But it might seem strange for a reader of the CPython documentation, who might not be a typing expert. We could probably just use the slightly-less-precise-but-simpler type `Any` instead.

Also: we need to use `typing.Any` if we want a snazzy link to the docs for `typing.Any`, the same as we have in the docs for the `args` parameter: https://docs.python.org/3/library/logging.html#logging.LogRecord. You can see in the Netlify docs preview that we don't currently have a link for `Any` in this parameter's type hint in the same way: https://deploy-preview-102875--python-cpython-preview.netlify.app/library/logging.html?highlight=logging#logging.LogRecord 

```suggestion
   :type msg: typing.Any
```

**Review by AlexWaygood — COMMENTED** [review]
Thanks!

File: Doc/library/logging.rst
Comment (aminalaee) [reply Doc/library/logging.rst]: I did not know about this, thanks :+1:

**Review by AlexWaygood — APPROVED** [review]
LGTM, thanks! I'll wait a few days before merging in case @vsajip has any objections, though, since he's the primary maintainer for `logging` :)

Here's a link directly to the docs preview for this class: https://deploy-preview-102875--python-cpython-preview.netlify.app/library/logging.html#logging.LogRecord

**Pull request merged**
